decoder/routes.go

```diff
@@ -0,0 +1,205 @@
+	err := db.GeneralDb.Get(&route,
+		`
+		SELECT *
+		FROM route
+		WHERE route.id = ?
+		`,
+		id,
+	)
+	if err == sql.ErrNoRows {
+		return nil, nil
+	} else if err != nil {
+		return nil, err
+	}
+
+	routeCache.Set(id, route, ttlcache.DefaultTTL)
+	return &route, nil
+}
+
+// hasChangesRoute compares two Route structs
+func hasChangesRoute(old *Route, new *Route) bool {
+	return old.Name != new.Name ||
+		old.Description != new.Description ||
+		old.DistanceMeters != new.DistanceMeters ||
+		old.DurationSeconds != new.DurationSeconds ||
+		old.EndFortId != new.EndFortId ||
+		!floatAlmostEqual(old.EndLat, new.EndLat, floatTolerance) ||
+		!floatAlmostEqual(old.EndLon, new.EndLon, floatTolerance) ||
+		old.Image != new.Image ||
+		old.ImageBorderColor != new.ImageBorderColor ||
+		old.Reversible != new.Reversible ||
+		old.StartFortId != new.StartFortId ||
+		!floatAlmostEqual(old.StartLat, new.StartLat, floatTolerance) ||
+		!floatAlmostEqual(old.StartLon, new.StartLon, floatTolerance) ||
+		old.Tags != new.Tags ||
+		old.Type != new.Type ||
+		old.Updated != new.Updated ||
+		old.Version != new.Version ||
+		old.Waypoints != new.Waypoints
+}
+
+func saveRouteRecord(db db.DbDetails, route *Route) error {
+	oldRoute, _ := getRouteRecord(db, route.Id)
+
+	if oldRoute != nil && !hasChangesRoute(oldRoute, route) {
+		return nil
+	}
+
+	if oldRoute == nil {
+		_, err := db.GeneralDb.NamedExec(
+			`
+			INSERT INTO route (
+			  id, name, description, distance_meters, 
+			  duration_seconds, end_fort_id, end_image, 
+			  end_lat, end_lon, image, image_border_color, 
+			  reversible, start_fort_id, start_image, 
+			  start_lat, start_lon, tags, type, 
+			  updated, version, waypoints
+			)
+			VALUES
+			  (
+				:id, :name, :description, :distance_meters, 
+				:duration_seconds, :end_fort_id, 
+				:end_image, :end_lat, :end_lon, :image, 
+				:image_border_color, :reversible, 
+				:start_fort_id, :start_image, :start_lat, 
+				:start_lon, :tags, :type, :updated, 
+				:version, :waypoints
+			  )
+			`,
+			route,
+		)
+
+		if err != nil {
+			return fmt.Errorf("insert route error: %w", err)
+		}
+	} else {
+		_, err := db.GeneralDb.NamedExec(
+			`
+			UPDATE route SET
+				name = :name,
+				description = :description,
+				distance_meters = :distance_meters,
+				duration_seconds = :duration_seconds,
+				end_fort_id = :end_fort_id,
+				end_image = :end_image,
+				end_lat = :end_lat,
+				end_lon = :end_lon,
+				image = :image,
+				image_border_color = :image_border_color,
+				reversible = :reversible,
+				start_fort_id = :start_fort_id,
+				start_image = :start_image,
+				start_lat = :start_lat,
+				start_lon = :start_lon,
+				tags = :tags,
+				type = :type,
+				updated = :updated,
+				version = :version,
+				waypoints = :waypoints
+			WHERE id = :id`,
+			route,
+		)
+
+		if err != nil {
+			return fmt.Errorf("update route error %w", err)
+		}
+	}
+
+	routeCache.Set(route.Id, *route, ttlcache.DefaultTTL)
```

Comment:
Not sure if we need a webhook?
Can be later, but I'm not sure anything would alert on a new route

Reply:
I do think that webhook would be useful, the same as for fort updates. But that should be something for another PR.